.teamcity/_self/projects/WebApp.kt

```diff
@@ -704,3 +705,103 @@ fun playwrightBuildType( targetDevice: String, buildUuid: String ): BuildType {
 		}
 	}
 }
+
+object PreReleaseE2ETests : BuildType({
+	id("Calypso_E2E_Pre_Release")
+	name = "Pre-Release E2E Tests"
+	description = "Runs a pre-release suite of E2E tests against trunk on staging, intended to be run after PR merge, but before deployment to production."
+
+	artifactRules = """
+		reports => reports
+		logs.tgz => logs.tgz
+		screenshots => screenshots
+	""".trimIndent()
+
+	vcs {
+		root(Settings.WpCalypso)
+		cleanCheckout = true
+	}
+
+	steps {
+		bashNodeScript {
+			name = "Prepare environment"
+			scriptContent = """
+				export NODE_ENV="test"
+				export PLAYWRIGHT_BROWSERS_PATH=0
+
+				# Install modules
+				${_self.yarn_install_cmd}
+
+				# Build packages
+				yarn workspace @automattic/calypso-e2e build
+			"""
+			dockerImage = "%docker_image_e2e%"
+		}
+		bashNodeScript {
+			name = "Run pre-release e2e tests"
+			scriptContent = """
+				shopt -s globstar
+				set -x
+
+				cd test/e2e
+				mkdir temp
+
+				export URL="https://wordpress.com"
```

Comment:
Maybe check with Systems that TeamCity gets the staging environment when it hits this url

Reply:
That's a good question! I will reach out to Systems to see if that's the case, and if not, how we can configure the needed TC agents to get there.

I'll go ahead and merge this PR for now so we can start testing with this build, and then we can come back and update this if we need to do a URL-based solution.